### PR TITLE
Added unit tests for sets and sequences

### PR DESCRIPTION
**Pull request by SamH532**

### CODE REVIEW

File: kazuki-core-test/src/test/kotlin/com/anaplan/engineering/kazuki/core/test/TestSequence.kt [hunk 1]
```diff
@@ -300,4 +300,14 @@ class TestSequence(
...
     }
+
+    @Test
+    fun filter() {
+        assertEquals(mk_Seq(1,2,3),mk_Seq(-3,-2,-1,0,1,2,3).filter{ e -> e > 0 })
+        assertEquals(mk_Seq(-3,-2,-1,0,1,2,3),mk_Seq(-3,-2,-1,0,1,2,3).filter{ e -> e > -4 })
+        assertEquals(mk_Seq(-3,-2,-1),mk_Seq(-3,-2,-1,0,1,2,3).filter{ e -> e < 0 })
+        assertEquals(mk_Seq(),mk_Seq(-3,-2,-1,0,1,2,3).filter{ e -> e > 3 })
+        assertEquals(mk_Seq<Int>(),mk_Seq<Int>().filter{ e -> e in mk_Seq(1) })
+    }
```
Comment (donbex) [inline kazuki-core-test/src/test/kotlin/com/anaplan/engineering/kazuki/core/test/TestSequence.kt]: ⛏️ Formatting could be improved.

File: kazuki-core-test/src/test/kotlin/com/anaplan/engineering/kazuki/core/test/TestSet.kt [hunk 2]
```diff
@@ -25,4 +24,64 @@ class TestSet {
...
+        assertEquals(3, mk_Set(mk_Set(1, 2), 2, 3).card)
+        assertEquals(0, mk_Set<Int>().card)
+    }
+
+    @Test
+    fun dunion() {
+        assertEquals(mk_Set<Int>(), dunion(mk_Set<Int>()))
+        assertEquals(mk_Set(1, 2, 3), dunion(mk_Set(mk_Set(1, 2), mk_Set(3))))
+        assertEquals(mk_Set(1, 2, 3), dunion(mk_Set(1, 2, 3)))
+        assertEquals(mk_Set<Int>(), dunion(mk_Set<Int>()))
```
Comment (donbex) [inline kazuki-core-test/src/test/kotlin/com/anaplan/engineering/kazuki/core/test/TestSet.kt]: 🚫 This is a duplicate of the test at line 63. However, you'll probably also want to test for the union of a non-empty set and the empty set.

**Review by donbex — CHANGES_REQUESTED** [review]

**Review by donbex — APPROVED** [review]
